server/services/area.service.js

```diff
@@ -154,11 +154,15 @@ export const getAllAreas = async ({ page = 1, limit = 10, ...query } = {}) => {
     areaOptions.shift();
   }
 
+  if (limit === 0) {
+    areaOptions.pop();
+  }
+
   const areas = await Area.aggregate(areaOptions);
 
   const total = getPaginationTotal(areas);
   const pagination = generatePagination(page, limit, total);
-  return { pagination, result: areas[0].data };
+  return { pagination, result: areas[0].data || areas };
```

Comment:
Lets define result as a variable here as done for pagination:

```js
const result = areas?.[0].data || areas
```

```suggestion
  return { pagination, result };
```